aws-android-sdk-appsync-compiler/src/main/kotlin/com/apollographql/apollo/compiler/GraphQLCompiler.kt

```diff
@@ -78,15 +78,7 @@ class GraphQLCompiler {
     return typeDeclarations.filter { it.kind == TypeDeclaration.KIND_SCALAR_TYPE }
         .associate { it.name to (this[it.name] ?: ClassNames.OBJECT.toString()) }
         .plus(idScalarTypeMap)
-        .plus("AWSDate" to ClassNames.STRING.toString())
-        .plus("AWSTime" to ClassNames.STRING.toString())
-        .plus("AWSDateTime" to ClassNames.STRING.toString())
         .plus("AWSTimestamp" to TypeName.LONG.box().toString())
```

Comment:
1. Why is this OK to retain while the other AWS-specific types are removed?
1. Why are we allowing custom overrides of AWS-defined classes? If we change some behavior in the future that relies on this being String, then we'd encounter issues?
1. What happens when this modified custom type gets passed to a newly generated schema that expects a String instead of a DateTime?
1. Would this use case not be solved by the customer defining their own custom DateTime type (e.g., `ConvertibleDateTime`) rather than overriding a core type?